### PR TITLE
Add sticker-pretrain

### PR DESCRIPTION
**Pull request by danieldk**
This change splits out support for pretraining into a separate
utility.

### CODE REVIEW

File: sticker-utils/src/bin/sticker-pretrain.rs [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        eprintln!(
+            "Epoch {} (validation): loss: {:.4}, acc: {:.4}, best epoch: {}, best acc: {:.4} {}",
+            epoch, loss, acc, best_epoch, best_acc, epoch_status
+        );
+    }
+
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
```
Comment (sebpuetz) [inline sticker-utils/src/bin/sticker-pretrain.rs:271]: Large lists of arguments are something that we see quite often in our tools. Maybe defining a struct that holds the relevant parameters could simplify some of the binaries?

File: sticker-utils/src/save.rs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        self.epoch
+    }
+
+    /// Save the model parameters when a save point has been reached.
+    pub fn save(&mut self, trainer: &TaggerTrainer, completed: CompletedUnit) -> Fallible<()> {
+        match completed {
+            CompletedUnit::Epoch => {
+                match self.schedule {
+                    SaveSchedule::Epoch | SaveSchedule::EpochAndBatches(_) => {
+                        trainer.save(format!("pretrain-epoch-{}", self.epoch))?
```
Comment (sebpuetz) [inline sticker-utils/src/save.rs]: Should the prefix `pretrain` be an argument? As far as I can tell, this is also the prefix for saved epochs when using `sticker-train`.

**Review by sebpuetz — CHANGES_REQUESTED** [review]

File: sticker-utils/src/save.rs (hunk 2 above)
Comment (danieldk) [reply sticker-utils/src/save.rs]: Good catch! will fix that tomorrow!

Comment (danieldk) [reply sticker-utils/src/save.rs]: Fixed.

File: sticker-utils/src/bin/sticker-pretrain.rs (hunk 1 above)
Comment (danieldk) [reply sticker-utils/src/bin/sticker-pretrain.rs:271]: I could do that in a separate PR, though I want to investigate first how ad-hoc the structs would be.

Comment (sebpuetz) [reply sticker-utils/src/bin/sticker-pretrain.rs:271]: Wrapping all the parameters in a struct and passing that wouldn't really alleviate anything, maybe it's possible to implement `run_epoch` on a struct. But that's nothing for this PR, i just saw the line while going through the changed files.

Comment (danieldk) [reply sticker-utils/src/bin/sticker-pretrain.rs:271]: Yeah, I already felt bad when I disabled that clippy lint ;).

**Comment by danieldk** [comment]
Ready to go?

**Comment by sebpuetz** [comment]
Didn't catch anything else when I went through the changes in the morning

**Pull request merged**